core/pkg/net/ssl/ssl.go

```diff
@@ -125,6 +127,56 @@ func AddOrUpdateCertAndKey(name string, cert, key, ca []byte) (*ingress.SSLCert,
 	}, nil
 }
 
+// AddOrUpdateCertAuth creates a .pem file with the specified CAs to be used in Cert Authentication
+func AddOrUpdateCertAuth(name string, ca []byte) (*ingress.SSLCert, error) {
+
+	caName := fmt.Sprintf("ca-%v.pem", name)
+	caFileName := fmt.Sprintf("%v/%v", ingress.DefaultSSLDirectory, caName)
+
+	tempCAPemFile, err := ioutil.TempFile(ingress.DefaultSSLDirectory, caName)
+	glog.V(3).Infof("AddOrUpdateCertAuth: Creating temp file %v for CA: %v", tempCAPemFile.Name(), caName)
+
+	if err != nil {
+		return nil, fmt.Errorf("could not create temp CA pem file %v: %v", tempCAPemFile.Name(), err)
+	}
+
+	_, err = tempCAPemFile.Write(ca)
+	if err != nil {
+		return nil, fmt.Errorf("could not write to CA pem file %v: %v", tempCAPemFile.Name(), err)
+	}
+
+	err = tempCAPemFile.Close()
+	if err != nil {
+		return nil, fmt.Errorf("could not close CA temp pem file %v: %v", tempCAPemFile.Name(), err)
+	}
+
+	pemCACerts, err := ioutil.ReadFile(tempCAPemFile.Name())
+	if err != nil {
+		return nil, err
+	}
+
+	pemCABlock, _ := pem.Decode(pemCACerts)
```

Comment:
Because we eat this also [here](https://github.com/rikatz/ingress/blob/dbdfbf2158da7571668aeb6b8d84c64741f4cb81/core/pkg/net/ssl/ssl.go#L69).

As this code is based on the very same function bellow, I didn't checked the validity of information. Will change here, and also in the function above.